crates/swift-integration-tests/src/result.rs

```diff
@@ -141,6 +146,30 @@ fn rust_func_return_result_unit_struct_opaque_rust(
     }
 }
 
+fn rust_func_returns_result_null_transparent_struct(
+    succeed: bool,
+) -> Result<(), ffi::ResultTestTransparentStruct> {
+    if succeed {
+        Ok(())
+    } else {
+        Err(ffi::ResultTestTransparentStruct("failed".to_string()))
+    }
+}
+
+impl std::error::Error for ffi::ResultTestTransparentStruct {}
+
+impl std::fmt::Debug for ffi::ResultTestTransparentStruct {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self)
```

Comment:
Let's avoid implying that these implementations are being used.

`unreachable!("Debug impl was added to pass type checking")`